packages/library-base/src/commonMain/kotlin/io/realm/Configuration.kt

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.internal.REPLACED_BY_IR
+import io.realm.log.LogLevel
+import io.realm.log.RealmLogger
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlin.reflect.KClass
+
+/**
+ * Configuration for log events created by a Realm instance.
+ */
+public data class LogConfiguration(
+    /**
+     * The [LogLevel] for which all log events of equal or higher priority will be reported.
+     */
+    public val level: LogLevel,
+
+    /**
+     * Any loggers to install. They will receive all log events with a priority equal to or higher than
+     * the value defined in [LogConfiguration.level].
+     */
+    public val loggers: List<RealmLogger>
+)
+
+interface Configuration {
+    // Public properties making up the RealmConfiguration
+    // TODO Add more elaborate KDoc for all of these
+    /**
+     * Path to the realm file.
+     */
+    public val path: String
+
+    /**
+     * Filename of the realm file.
+     */
+    public val name: String
+
+    /**
+     * The set of classes included in the schema for the realm.
+     */
+    public val schema: Set<KClass<out RealmObject>>
+
+    /**
+     * The log configuration used for the realm instance.
+     */
+    public val log: LogConfiguration
+
+    /**
+     * Maximum number of active versions.
+     *
+     * Holding references to objects from previous version of the data in the realm will also
+     * require keeping the data in the actual file. This can cause growth of the file. See
+     * [Builder.maxNumberOfActiveVersions] for details.
+     */
+    public val maxNumberOfActiveVersions: Long
+
+    /**
+     * The schema version.
+     */
+    public val schemaVersion: Long
+
+    /**
+     * 64 byte key used to encrypt and decrypt the Realm file.
+     *
+     * @return null on unencrypted Realms.
+     */
+    val encryptionKey: ByteArray?
+
+    /**
+     * This class contains shared properties across the two types of configuration builders.
```

Comment:
This level of details doesn't belong to a Javadoc level IMO, we can keep it as a classic comment though